### PR TITLE
Fetch config from Tools API

### PR DESCRIPTION
**Pull request by jyothishnt**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1492

## Description
Fetch blast config from tools api

## Deployment URL
http://fetch-config.review.ensembl.org/

## Views affected
Blast

### CODE REVIEW

**Comment by jyothishnt** [comment]
Ready for review, but review app fails due to pipeline failuee (issues at webprod)

File: src/content/app/tools/blast/state/blast-api/blastApiSlice.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// import { setBlastSettingsConfig } from '../blast-form/blastFormSlice';
+
+import type { BlastSettingsConfig } from 'src/content/app/tools/blast/types/blastSettings';
+
+const blastApiSlice = restApiSlice.injectEndpoints({
+  endpoints: (builder) => ({
+    blastConfig: builder.query<BlastSettingsConfig, void>({
+      query: () => ({
+        url: `api/tools/blast/config`
```
Comment (azangru) [inline src/content/app/tools/blast/state/blast-api/blastApiSlice.ts:27]: we want to get the `api/tools` part from our config file (not blast config, but our app config 😄):

```
import config from 'config';

...

url: `${config.toolsApiBaseUrl}/blast/config`

```

File: src/content/app/tools/blast/state/blast-api/blastApiSlice.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import restApiSlice from 'src/shared/state/api-slices/restSlice';
+
+// import { setBlastSettingsConfig } from '../blast-form/blastFormSlice';
```
Comment (azangru) [inline src/content/app/tools/blast/state/blast-api/blastApiSlice.ts:19]: ✂️

File: src/shared/state/api-slices/restSlice.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export default createApi({
+  reducerPath: 'restApi',
+  baseQuery: fetchBaseQuery({ baseUrl: '/' }),
```
Comment (azangru) [inline src/shared/state/api-slices/restSlice.ts:21]: You will notice that there's a warning in the terminal:

```
Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.
```

Let's provide the `fetch` from the `cross-fetch` package that we have as a `fetchFn` option:

```
import fetch from 'cross-fetch'

...

baseQuery: fetchBaseQuery({ baseUrl: '/', fetchFn: fetch }),
```

**Comment by azangru** [comment]
**Note:** we need to make sure that @veidenberg has updated the config with [these changes](https://github.com/Ensembl/ensembl-client/pull/711/files#diff-5380e6b6fdf52698462377b1d34e7accb2e2efdeee3c0c8cbc01a32909ddf8ce) before merging this PR.

**Comment by veidenberg** [comment]
Tools API endpoint updated with the new config.

**Review by azangru — APPROVED** [review]

**Pull request merged**